src/analysis/liouville/liouville_constant.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa, Jujian Zhang
+-/
+import analysis.liouville.basic
+/-!
+
+# Liouville constants
+
+This file contains a construction of a family of Liouville numbers.
+The most important property is that they are examples of transcendental real numbers.
+This fact is recorded in `is_liouville.is_transcendental_of_liouville_constant` (a result in
+a subsequent PR).
+-/
+
+noncomputable theory
+open_locale nat big_operators
+open set real finset
+
+section m_is_real
+
+variable {m : ℝ}
+
+namespace liouville
+
+/--
+For a real number `m`, Liouville's constant is
+$$
+\sum_{i=0}^\infty\frac{1}{m^{i!}}.
+$$
+The series converges only for `1 < m`.  However, there is no restriction on `m`, since,
+if the series does not converge, then the sum of the series is defined to be zero.
+-/
+def liouville_number (m : ℝ) := ∑' (i : ℕ), 1 / m ^ i!
```

Comment:
Here and below: 
```suggestion
def liouville_number (m : ℝ) : ℝ := ∑' (i : ℕ), 1 / m ^ i!
```
The type is fairly obvious for these `def`s, but I think making them explicit is generally good practice.